src/main/java/org/opentripplanner/middleware/bugsnag/jobs/BugsnagEventHandlingJob.java

```diff
@@ -142,4 +164,29 @@ private void removeStaleEvents() {
         );
         Persistence.bugsnagEvents.removeFiltered(Filters.lte("receivedAt", reportingWindowCutoff));
     }
+
+    /**
+     * Extract Bugsnag project error from webhook delivery.
+     */
+    public static void processWebHookDelivery(Request request) {
+        if (BUGSNAG_WEBHOOK_PERMITTED_IPS == null) {
+            LOG.warn("Bugsnag webhook permitted IPs not defined. Caller IP not validated nor content processed.");
+            return;
+        } else if(!BUGSNAG_WEBHOOK_PERMITTED_IPS.contains(request.ip())) {
+            LOG.warn("Bugsnag webhook delivery called from unauthorized IP: {}. Request rejected.", request.ip());
+            return;
+        }
+        try {
+            BugsnagWebHookDelivery webHookDelivery =
+                JsonUtils.getPOJOFromJSON(request.body(), BugsnagWebHookDelivery.class);
+            if (webHookDelivery != null) {
+                LOG.info("New event delivered via the Bugsnag webhook. Storing and notifying subscribed admin users.");
+                Persistence.bugsnagEvents.create(new BugsnagEvent(webHookDelivery));
+                // Notify any subscribed users about new events.
+                sendEmailForEvents(1);
+            }
+        } catch (JsonProcessingException e) {
+            BugsnagReporter.reportErrorToBugsnag("Failed to parse webhook delivery!", request.body(), e);
+        }
+    }
```

Comment:
This class is specifically for a Job, so I think putting this webhook handler in here is adding too many responsibilities to this class. See proposed refactor in https://github.com/ibi-group/otp-middleware/pull/151.